lib/plugins/aws/package/lib/save-service-state.js

```diff
@@ -23,6 +23,8 @@ module.exports = {
       _.omit(this.serverless.service, ['serverless', 'package'])
     );
     const selfReferences = findReferences(strippedService, this.serverless.service);
+
+    // #7747, in testing `selfReferences` is always an empty array (?), this likely needs refactoring
     selfReferences.forEach((refPath) => _.set(strippedService, refPath, '${self:}'));
```

Comment:
Yes, this can be removed. It was overlooked with v3 release. Ideally if you can propose another PR with that